Topics/Tech_Stacks/Vue_Component_Setup_and_Tips

```diff
@@ -0,0 +1,73 @@
+
+Components are how it's all done. Every .vue file is a Vue Component! In these files, you can condense the 3 key aspects of a webpage, the HTML, code-behind, and CSS styling. This centralization and user-friendliness has made Vue renowned for its simplicity.
+
+**Am I quoting too much from official documentation? Should I put my own spin on it or cut out general info about the Vue.js framework?**
+
+> To learn more, please visit the [Vue website](https://vuejs.org/guide/introduction.html).
+
+## Format of Vue Components<a name="implementation"></a>
+Here is the general format of a Vue Component.
+
+
+```c
+# Example code
+<script setup>
+import { ref } from 'vue'
+const count = ref(0)
+</script>
+
+<template>
+  <button @click="count++">Count is: {{ count }}</button>
+</template>
+
+<style scoped>
+button {
+  font-weight: bold;
+}
+</style>
+```
+The top is the Javascript code-behind, the middle is the HTML, and the bottom is the CSS style. Nice and centralized. Thankfully, the order doesn't matter. However, as the script is the most complicated, we'll talk about it last.
+
+**Should I explain what the example code above means? Like, what import {ref} does? Or maybe I should give a different example, one similar to what we worked on in our CSC301 project? I worry it might be too long**
+
+### HTML Template
+Ever seen the HTML of a webpage? That and Vue's HTML template are almost identical. Just remember, the root container must be labelled as ```<template>```, and it can only have one container inside that. Inside *that* sub-container though, anything is possible! On top of things like your standard ```<div>```, you can also use **other** Vue Components that you've imported in the script section! More on that in a bit.
+
+Here, I'm going to *only* show how Vue HTML components have if-else visibility features, and the ability to insert components through built-in for loops. Am I missing anything important to do with HTML in Vue?
+
+### CSS
+The CSS styling in Vue works as it does anywhere else, but with one key difference. See that ```scoped``` attribute in the ```<style>``` tag? When the tag has that attribute, the CSS can only apply to the component it's present in. Removing it however, allows for usage anywhere else in the program. Just...be careful about CSS conflicts and shadowing.
+
+### Script
+This is the meat of Vue. This is where you essentially define the webpage as an Object-Oriented-Programming Class! It can have it's own methods, attributes, as well as import scripts and other components for use! It's a very dense part of learning Vue, but here are the basics.
+
+Here I'm going to explain how to mount the app, basically talking about props, the default{} variables, computed methods, import statements, etc. It's going to be a rather dense section, as I have to talk about class variables and how mounted() works at runtime. Should I go into asynchronous methods/components?
+
```

Comment:
I think keep it as a quick primer for now -- this looks good -- but skip the async stuff. I think your goal is a basic guide, per our conversations earlier?